activerecord/lib/active_record/railties/databases.rake

```diff
@@ -77,6 +77,8 @@ db_namespace = namespace :db do
   namespace :migrate do
     # desc  'Rollbacks the database one migration and re migrate up (options: STEP=x, VERSION=x).'
     task redo: [:environment, :load_config] do
+      raise "Empty VERSION provided" if !ENV["VERSION"].nil? && ENV["VERSION"].empty?
```

Comment:
Usually in ruby you don't need to check for `nil?` unless `false` have a different meaning, but in this case `nil` and `false` needs to have the same behavior so it is better to not check for `nil?`